### DISCUSSION TITLE
Why does shap.plot.heatmap use mean to calculate the f(x) value

### DISCUSSION THREAD

**Discussion by sroener** [shap/shap#3544] [Q&A] [answered]
Hi,

I stumbled upon the following code snippet:

```python
fx = values.T.mean(0)
```

in the [heatmap plotting function](https://github.com/shap/shap/blob/dbb275978abe27def1afe73fdd097d071c7fbe38/shap/plots/_heatmap.py#L141C1-L146C6)

and was wondering why mean was used instead of sum.

From my understanding, shap values are additive and are interpreted in relation to the base_value (expected value) as offset. 

I checked, it seems not to have an effect on the resulting plot, because the values are scaled using the abs_max method.

**Comment by connortann** [comment, marked as answer]
Interesting question. As you say, both operations lead to the same result: they are only different by a constant factor, and the values are then rescaled. However, I agree with you that `sum` would be a bit more logical than `mean` as it better fits with the interpretation of shap values.

**Reply by sroener** [reply]
Thank you for the answer.

I propose to change it to `sum` during the API changes discussed in #3411 to make it more consistent with the general idea of how SHAP works. That way, confusion in future development can be avoided. I will include the change in a future pull request.

**Reply by connortann** [reply]
A pull request would be most welcome!
